src/alpaca_ast_gen.erl

```diff
@@ -437,13 +441,15 @@ rename_bindings(Environment, #alpaca_binding{}=TopLevel) ->
     end.
 
 rebind_args(#env{current_module=Mod}=Env, Map, Args) ->
-    F = fun({symbol, L, N}, {#env{next_var=NV}=E, AccMap, Syms}) ->
+    F = fun({'Symbol', _}=S, {#env{next_var=NV}=E, AccMap, Syms}) ->
+                N = alpaca_ast:symbol_name(S),
+                L = alpaca_ast:line(S),
                 case maps:get(N, AccMap, undefined) of
                     undefined ->
-                        Synth = next_var(NV),
+                        Synth = unicode:characters_to_binary(next_var(NV), utf8),
```

Comment:
You could move the list -> binary translation into next_var, since all callers of next_var now make that conversion.